scipy/signal/_peak_finding_utils.pyx

```diff
@@ -74,3 +74,70 @@ def _argmaxima1d(np.float64_t[:] x not None):
 
     maxima.resize(m, refcheck=False)  # Keep only valid part of array memory.
     return maxima
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _select_by_peak_distance(np.intp_t[:] peaks not None,
+                             np.float64_t[:] priority not None,
+                             dmin):
```

Comment:
I noticed that in the Jupyter nb you linked in this PR, `dmin` is actually typed as `np.int64_t` -- is there a compelling reason to drop that typing in the PR itself?

Reply:
Perhaps you've dropped the original typing to allow for a later processing of a more flexible input with the ceiling operation below, but can't you still type a `double` or something of that nature?

Reply:
Yeah, the reasoning was, that the user may want to specify the minimal distance as a fraction / multiple of the sampling frequency which wouldn't necessarily yield an int. That way the user doesn't have to include an `int(round(...))` in the function signature of `find_peaks`.

I could have handled this conversion in `find_peaks`, prior to this function, or like now. The second way felt more natural to me (the old version didn't need a round-type-casting construct to accept dmin, too). That's the reason for dropping `np.float64_t`. It's still in the notebook because I simply forgot to change it there as well.